src/v/kafka/server/handlers/delete_topics.cc

```diff
@@ -125,6 +143,13 @@ delete_topics_handler::handle(request_context ctx, ss::smp_service_group) {
         });
     }
 
+    for (auto& topic : nodelete_topics) {
+        resp.data.responses.push_back(deletable_topic_result{
+          .name = std::move(topic),
+          .error_code = error_code::topic_authorization_failed,
```

Comment:
what do you think about:

1. logging a message here to announce that deletion is prevented because it is a topic protected via a mechanism that is different than normal ACLs because the experience at the client side will be identical.
2. the deltable_topic_result also has an optional `std::optional<ss::sstring> error_message{};` you can fill in which clients might use to provide more context in error messages.

Reply:
there is also another kafka error code `error_code::topic_deletion_disabled` which might be more appropriate? i think it is used in kafka when all topic deletion is disabled. i don't think it matters much, but something to think about. Travis B may have some insight in case clients handle these error codes differently. I would suggest pinging him about which error would likely behave the most reasonably.

Reply:
1. Will add logging when an operation isn't authorized due to the `kafka_noproduce_topics` or `kafka_nodelete_topics`.
2. Will message Travis B about this and `error_code::topic_deletion_disabled` and see what may be best from a client's perspective.

Reply:
Travis B. got back to me and said that `topic_authorization_failed` is probably what we want to return here.

Reply:
thanks for looking into the details